src/rules/abilities.clj

```diff
@@ -1,39 +1,46 @@
 (ns rules.abilities)
 
-(defn add-power
-  "Creates the ability function"
+(defn ^:private row-affinity
+  [row-type increase]
+  (fn
+;   "When played in a row-type row,
+;   alters a its power, by adding the a defined value to it.
+;   This power may be negative, resulting in a decrease"
+    [game-state play]
+    (if (= (get-in game-state [:rows (:row-id play) :type])
+           row-type)
+      (update-in
+        game-state
+        [:cards (:card-id play) :power]
+        #(+ % increase))
+      game-state)))
+
+(defn ^:private strengthen
   [increase]
-  {:ability
-   (fn    
-;    "Alters a cards' power, by adding the a defined value to it.
-;    This power may be negative, resulting in a decrease"
-     [game-state play]
-     (update-in
-       game-state
-       [:cards (:target play) :power]
-       #(+ % increase)))
-   :description (if (neg? increase)
-                    (str "Weaken " (- increase))
-                    (str "Enhance " increase))
-   :target 1})
+  (fn    
+;   "Alters a cards' power, by adding the a defined value to it.
+;   This power may be negative, resulting in a decrease"
+    [game-state play]
+    (update-in
+      game-state
+      [:cards (:target play) :power]
+      #(+ % increase))))
 
-(defn add-power-on-row-type
-  "Creates the ability function"
-  [row-type increase]
-  {:ability
-   (fn
-;    "When played in a row-type row,
-;    alters a its power, by adding the a defined value to it.
-;    This power may be negative, resulting in a decrease"
-     [game-state play]
-     (if (= (get-in game-state [:rows (:row-id play) :type])
-            row-type)
-       (update-in
-         game-state
-         [:cards (:card-id play) :power]
-         #(+ % increase))
+(defn ^:private weaken
+  "Same as strengthen but negative"
+  [decrease]
+  (strengthen (- decrease)))
+
+(defn generate-ability-fn
+  "Creates an ability function from its description"
+  [ability-description]
+  (let [ability-generator (ns-resolve 'rules.abilities (symbol (name (first ability-description))))
```

Comment:
I feel like `ns-resolve` is effectively obfuscation.

`(let [ability-generator (get ability-list (first ability-description))]` seems much more readable